metagraph/src/graph/representation/succinct/boss.cpp

```diff
@@ -849,9 +849,12 @@ bool BOSS::compare_node_suffix(edge_index first, const TAlphabet *second) const
 
 /**
  * Given an edge index i, this function returns the k-mer sequence of its
- * source node.
+ * source node, and the node whose last character corresponds to the first
+ * character of the sequence. If the graph is suffix indexed, then the returned
+ * node is the last node visited after k - indexed_suffix_length_ bwd steps.
  */
-std::vector<TAlphabet> BOSS::get_node_seq(edge_index x) const {
+std::tuple<std::vector<TAlphabet>, edge_index, bool> BOSS
+::get_node_seq_with_end_node_indexed(edge_index x) const {
```

Comment:
The bool is essentially used to indicate that the returned edge_index is unusable.

So, just return npos for edge_index in that case, and you don't need to add this bool value.
```suggestion
std::tuple<std::vector<TAlphabet>, edge_index> BOSS
::get_node_seq_with_end_node_indexed(edge_index x) const {
```

Reply:
Never mind, this whole thing should be changed.

See another comment for `get_node_seq_with_end_node`